alert-common/src/main/java/com/synopsys/integration/alert/common/action/CustomFunctionAction.java

```diff
@@ -22,28 +22,53 @@
  */
 package com.synopsys.integration.alert.common.action;
 
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
 import org.springframework.http.HttpStatus;
 import org.springframework.web.server.ResponseStatusException;
 
+import com.github.jsonldjava.shaded.com.google.common.base.Predicate;
+import com.synopsys.integration.alert.common.descriptor.DescriptorProcessor;
+import com.synopsys.integration.alert.common.descriptor.config.field.ConfigField;
+import com.synopsys.integration.alert.common.descriptor.config.field.errors.AlertFieldStatus;
+import com.synopsys.integration.alert.common.descriptor.config.field.errors.FieldStatusSeverity;
+import com.synopsys.integration.alert.common.descriptor.config.field.validation.FieldValidationUtility;
 import com.synopsys.integration.alert.common.rest.HttpServletContentWrapper;
 import com.synopsys.integration.alert.common.rest.ResponseFactory;
 import com.synopsys.integration.alert.common.rest.model.FieldModel;
 import com.synopsys.integration.alert.common.security.authorization.AuthorizationManager;
+import com.synopsys.integration.alert.common.util.DataStructureUtils;
 import com.synopsys.integration.exception.IntegrationException;
 
 public abstract class CustomFunctionAction<T> {
     public static final String API_FUNCTION_URL = "/api/function";
-    private AuthorizationManager authorizationManager;
 
-    public CustomFunctionAction(AuthorizationManager authorizationManager) {
+    private final String fieldKey;
+    private final AuthorizationManager authorizationManager;
+    private final DescriptorProcessor descriptorProcessor;
+    private final FieldValidationUtility fieldValidationUtility;
+
+    public CustomFunctionAction(String fieldKey, AuthorizationManager authorizationManager, DescriptorProcessor descriptorProcessor, FieldValidationUtility fieldValidationUtility) {
+        this.fieldKey = fieldKey;
         this.authorizationManager = authorizationManager;
+        this.descriptorProcessor = descriptorProcessor;
+        this.fieldValidationUtility = fieldValidationUtility;
     }
 
     public ActionResponse<T> createResponse(FieldModel fieldModel, HttpServletContentWrapper servletContentWrapper) {
```

Comment:
This function is the primary change, most of the other changes are related to updating autowired constructor params.